### PR TITLE
adding/updating security groups for publisher and publishing-api

### PR DESCRIPTION
**Pull request by fredericfran-gds**
- workers of publisher and publishing-api needs access to redis as they use sidekiq.

- adding egress rule for publisher worker and publishing-api

### CODE REVIEW

File: terraform/deployments/govuk-publishing-platform/security_group_rules.tf [hunk 1]
```diff
@@ -261,10 +261,40 @@ resource "aws_security_group_rule" "publisher_to_any_any" {
...
+  protocol          = -1
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = module.publisher_worker.security_group_id
+}
+
 #
 # Publishing API
 #
 
+resource "aws_security_group_rule" "publishing_web_to_any_any" {
```
Comment (karlbaker02) [inline terraform/deployments/govuk-publishing-platform/security_group_rules.tf:278]: This is for Publishing API, can we rename it something like `publishing_api_web_to_any_any`?

File: terraform/deployments/govuk-publishing-platform/security_group_rules.tf [hunk 2]
```diff
@@ -261,10 +261,40 @@ resource "aws_security_group_rule" "publisher_to_any_any" {
...
+  description       = "Publishing web sends requests to anywhere over any protocol"
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = -1
+  cidr_blocks       = ["0.0.0.0/0"]
+  security_group_id = module.publishing_api_web.security_group_id
+}
+
+resource "aws_security_group_rule" "publishing_worker_to_any_any" {
```
Comment (karlbaker02) [inline terraform/deployments/govuk-publishing-platform/security_group_rules.tf:288]: Similarly this is for Publishing API, can we rename it something like `publishing_api_worker_to_any_any`?

Comment (fredericfran-gds) [reply terraform/deployments/govuk-publishing-platform/security_group_rules.tf:288]: yes.

File: terraform/deployments/govuk-publishing-platform/security_group_rules.tf (hunk 1 above)
Comment (fredericfran-gds) [reply terraform/deployments/govuk-publishing-platform/security_group_rules.tf:278]: yes!

**Review by karlbaker02 — APPROVED** [review]
Looks good Fred! 💯

**Pull request merged**